updated figure test text (#3681)
sunpy/sunpy: docs/dev_guide/tests.rst

    def test_simple_plot():
        plt.plot([0,1])

The current figure at the end of the unit test, or an explicitly returned figure, has its hash (currently ``SHA256``) compared against an established hash collection (more on this below).
If the hashes do not match, the figure has changed, and thus the test is considered to have failed.

To run the figure tests you need to be very careful, as any pixel that has changed, will change the hash.

> In order to avoid changes due to package versions, we recommend using tox::

Reviewer: ```suggestion
In order to avoid changes due to different package versions, we recommend using tox::
```

    $ tox -e figure

> This will ensure that any figures created are checked using the current packages we test against.

Reviewer: ```suggestion
This will ensure that any figures created are checked using the package versions that were used to create the original figure hashes.
```

> Running this will create a folder, "figure_test_images", within your work folder ("<local clone location>/figure_test_images") and this is ignored by git.

Reviewer: ```suggestion
Running this will create a folder, "figure_test_images", within your work folder ("<local clone location>/figure_test_images"), which is ignored by git.
```

> Inside this folder will be all the images created as well as a json file with the hashes of the figures created by the test run.

Reviewer: ```suggestion
Inside this folder will be all the images created, as well as a json file with the hashes of the figures created by the test run.
```

> So if you have created a new figure test, by running the tox command, it will create the figure and the corresponding hash for you to use.

Reviewer: This seems to repeat the sentence above?

Author: I'll delete it.

> The current hashes are located within "sunpy/tests/figure_tests_env_py36.json" and this will be where you will update old hashes or create new figure entries.

Reviewer: ```suggestion
The current hashes are located within "sunpy/tests/figure_tests_env_py36.json" and this will be where you will need to update old hashes or create new figure entries if anything changes.
```

> If you are adding a new figure test, you will need to update this `repository <https://github.com/sunpy/sunpy-figure-tests>`__ that stores the current figure tests, we use this for a visual comparison of figure tests.

Reviewer: ```suggestion
If you are adding a new figure test, you will also need to update this `repository <https://github.com/sunpy/sunpy-figure-tests>`__ that stores the current figure tests, which we use for a visual comparison of figure tests.
```

> The current figure at the end of the unit test, or an explicitly returned figure, has its hash compared against an established hash library (more on this below).

Reviewer 2: We should also mention that the hashing algorithm used is `SHA256` and the hash is stored as a `HEX` string.

Author: I added the algorithm.
